sklearn/tree/_criterion.pyx

```diff
@@ -245,16 +248,11 @@ cdef class Criterion:
                 (sum_left <= upper_bound * weighted_n_left) &
```

Comment:
I can't comment on line 237, but can we early return `1` when `monotonic_cst == 0`?